File: VB_NET/Cap08/c08ex10.vb
' c08ex10.vb

Imports System
Imports System.Console

Module Program
    Function FibonacciBase(ByVal N As Integer, ByVal X As Integer, _
        ByVal Y As Integer) As Integer
        If (N = 0) Then
            Return X
        End If
        If (N = 1) Then
            Return Y
        End If
        If (N >= 2) Then
            Return FibonacciBase(N - 1, Y, X + Y)
        End If
        Return 0
    End Function

    Function Fibonacci(ByVal N As Integer) As Integer
        Return FibonacciBase(N, 0, 1)
    End Function

    Sub Main(ByVal args() As String)
        For I As Integer = 0 To 14
            WriteLine($"{I} - {Fibonacci(I)}")
        Next
        Write(vbCrLf & "Pressione <Enter> para sair... ")
        ReadLine()
    End Sub
End Module

File: RUBY/Cap08/c08ex10.rb
# c08ex10.rb

#!/usr/bin/env ruby

def fibonacci_base(n, x, y)
  if (n == 0)
    return x
  end
  if (n == 1)
    return y
  end
  if (n >= 2)
    return fibonacci_base(n - 1, y, x + y)
  end
end

def fibonacci(n)
  fibonacci_base(n, 0, 1)
end

for i in 0..14
  printf "%2d - %d\n", i, fibonacci(i)
end

print "\nTecle <Enter> para encerrar... "
gets
